pkg/apis/core/v1/handler.go

```diff
@@ -322,16 +322,114 @@ func (h *handler) DeleteCluster(request *restful.Request, response *restful.Resp
+	if err != nil {
+		restplus.HandleInternalError(response, request, err)
+		return
+	}
+	for _, cronBackup := range cronBackups.Items {
+		if cronBackup.Spec.ClusterName == c.Name {
+			// make the cronBackup disable
+			if _, ok := cronBackup.Labels[common.LabelCronBackupEnable]; ok {
+				delete(cronBackup.Labels, common.LabelCronBackupEnable)
+				cronBackup.Labels[common.LabelCronBackupDisable] = ""
+				_, err = h.clusterOperator.UpdateCronBackup(request.Request.Context(), &cronBackup)
+			}
+			// delete cronBackup
+			if err = h.clusterOperator.DeleteCronBackup(request.Request.Context(), cronBackup.Name); err != nil {
+				restplus.HandleInternalError(response, request, err)
+				return
+			}
+		}
+	}
+
+	// delete backup before delete cluster
 	q := query.New()
 	q.LabelSelector = fmt.Sprintf("%s=%s", common.LabelClusterName, c.Name)
-	backups, err := h.clusterOperator.ListBackupEx(request.Request.Context(), q)
+	backups, err := h.clusterOperator.ListBackups(request.Request.Context(), q)
 	if err != nil {
 		restplus.HandleInternalError(response, request, err)
 		return
 	}
-	if backups.TotalCount > 0 {
-		restplus.HandleInternalError(response, request, fmt.Errorf("before deleting the cluster, please delete the cluster backup file first"))
-		return
+	for _, backup := range backups.Items {
+		bp, err := h.clusterOperator.GetBackupPointEx(request.Request.Context(), c.Labels[common.LabelBackupPoint], "0")
+		if err != nil {
+			restplus.HandleInternalError(response, request, err)
+			return
+		}
+
+		if backup.PreferredNode != "" {
+			_, err = h.clusterOperator.GetNodeEx(request.Request.Context(), backup.PreferredNode, "0")
+			if err != nil {
+				restplus.HandleInternalError(response, request, err)
+				return
+			}
+		}
+
+		// create operation
+		op := &v1.Operation{}
+		op.Name = uuid.New().String()
+		op.Labels = make(map[string]string)
+		op.Labels[common.LabelOperationAction] = v1.OperationDeleteBackup
+		op.Labels[common.LabelTimeoutSeconds] = strconv.Itoa(v1.DefaultBackupTimeoutSec)
+		op.Labels[common.LabelClusterName] = c.Name
+		op.Labels[common.LabelBackupName] = backup.Name
+		op.Labels[common.LabelTopologyRegion] = c.Masters[0].Labels[common.LabelTopologyRegion]
+		op.Status.Status = v1.OperationStatusRunning
+
+		if backup.Status.ClusterBackupStatus == v1.ClusterBackupRestoring || backup.Status.ClusterBackupStatus == v1.ClusterBackupCreating {
+			restplus.HandlerErrorWithCustomCode(response, request, http.StatusExpectationFailed, 417, "", fmt.Errorf("backup is %s now, can't delete", backup.Status.ClusterBackupStatus))
+			return
+		}
+
+		steps := make([]v1.Step, 0)
+		var actBackup *k8s.ActBackup
+		meta := component.ExtraMetadata{
+			ClusterName: c.Name,
+		}
+		meta.Masters = append(meta.Masters, component.Node{ID: backup.PreferredNode})
+		ctx := component.WithExtraMetadata(context.TODO(), meta)
+
+		switch bp.StorageType {
```

Comment:
The extra actback is used, and it is recommended to wrap it into a generic function